### PR TITLE
Adding reader options for Excel: header & skiprows

### PR DESCRIPTION
**Pull request by agarctfi**
## What

This PR resolves issues with the `openpyxl_chunk_reader` method in the `source-file` connector to ensure the `reader_options` (`names`, `header`, and `skiprows`) work as expected. We use `openpyxl_chunk_reader`  for excel files to use its lazy loading feature to solve for OOM issues related to large Excel files(https://github.com/airbytehq/airbyte/pull/25575). Due to this, pandas reader options need to be manually implemented. 

Fixes #<Insert-GitHub-Issue-Link> (if applicable).

---

## How

- Updated the `openpyxl_chunk_reader` to:
  - Properly handle the `names` option by overriding column names.
  - Correctly use the `header` option to extract and apply the specified row as headers.
  -  Handle the `skiprows` option to skip rows on the Excel file specified by the user.
  - Refactored logic to align with expected behavior while maintaining existing chunking functionality.

---

## Review guide

1. **`client.py`**
   - See changes to the `openpyxl_chunk_reader` method.
   - Focus on the handling of `reader_options` and edge case logic.
2. **`unit_tests/test_client.py`**
   - Review the tests for `names`, `header`, and `skiprows`.
   - Check edge case tests.

---

### End Result:
- Users can now reliably use `names`, `header`, and `skiprows` options when processing Excel files.
- Meaningful error messages are returned for invalid configurations or insufficient data.

### Negative Side Effects:
- None identified. Changes maintain existing functionality while addressing bugs and improving edge case handling.

---

## Can this PR be safely reverted and rolled back?

- [X] YES 💚
- [] NO ❌

### CODE REVIEW

**Comment by agarctfi** [comment]
Regression Tests [https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/manu[…]32b55875073ebd257c4b5267b/source-file/0.5.16/output.html](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/manual/alfredo_source-file_copy-add-excel-reader_options/1733509141/da98ca3288039d332b55875073ebd257c4b5267b/source-file/0.5.16/output.html)

**Review by aldogonzalez8 — APPROVED** [review]
APPROVED

File: docs/integrations/sources/file.md [hunk 1]
```diff
@@ -298,6 +298,7 @@ In order to read large files from a remote location, this connector uses the [sm
 
 | Version | Date       | Pull Request                                             | Subject                                                                                                 |
 | :------ | :--------- | :------------------------------------------------------- | :------------------------------------------------------------------------------------------------------ |
+| 0.5.16 | 2024-12-04 | [48804](https://github.com/airbytehq/airbyte/pull/48804) | Added reader options: skiprows & header for Excel files |
```
Comment (aldogonzalez8) [inline docs/integrations/sources/file.md:301]: nit: you may just need to update this date.

**Pull request merged**